it/src/it/simple/verify.groovy

```diff
@@ -23,4 +24,11 @@ import io.fabric8.maven.it.Verify
           new File(basedir, sprintf("/expected/%s.yml",it)))
 }
 
+Map selector = Verify.readWithPath(
+        new File(basedir,"/target/classes/META-INF/fabric8/kubernetes/fabric8-maven-sample-zero-config-deployment.yml"),
+        "spec.selector.matchLabels")
+
+assertNotNull(selector)
+assertNull(selector.get("version"))
```

Comment:
Actually, in order to allow for robust tests, we can't check for required missing fields. We just ignore all extra fields (assuming that we typically want to check that something was added properly). So I think that this trade off is ok.